samples/unit-tests/indicator-macd/recalculations/demo.js

```diff
@@ -488,3 +488,78 @@ QUnit.test('Test algorithm on data updates.', function (assert) {
     chart.series[1].remove();
     assert.ok(true, 'No error when removing MACD without lines (#8848).');
 });
+
+QUnit.test('After changing the shortPeriod all points should be visible, #14197.', function (assert) {
+    const chart = Highcharts.stockChart('container', {
+            yAxis: [
+                {
+                    height: '50%'
+                },
+                {
+                    top: '60%',
+                    height: '40%'
+                }
+            ],
+            series: [
+                {
+                    id: 'main',
+                    data: [
+                        459.99,
+                        448.85,
+                        446.06,
+                        450.81,
+                        442.8,
+                        448.97,
+                        444.57,
+                        441.4,
+                        430.47,
+                        420.05,
+                        431.14,
+                        425.66,
+                        430.58,
+                        431.72,
+                        437.87,
+                        428.43,
+                        428.35,
+                        432.5,
+                        443.66,
+                        455.72,
+                        454.49,
+                        452.08,
+                        452.73,
+                        461.91,
+                        463.58,
+                        461.14,
+                        452.08,
+                        442.66,
+                        428.91,
+                        429.79,
+                        431.99,
+                        427.72,
+                        423.2,
+                        426.21,
+                        426.98,
+                        435.69,
+                        434.338
+                    ]
+                }, {
+                    yAxis: 1,
+                    type: 'macd',
+                    linkedTo: 'main',
+                    params: {
+                        shortPeriod: 10,
+                        longPeriod: 26,
+                        signalPeriod: 9,
+                        period: 26
+                    }
+                }
+            ]
+        }),
+        points = chart.series[1].points;
+
+    assert.strictEqual(
+        points[points.length - 1].category,
+        36,
```

Comment:
It would be better to check if last point's `x` matches main series last point, for example: `series[0].points[points.length - 1].category`.

Instead of `point.category` you can use `point.x`